server/src/servershot.js

```diff
@@ -8,24 +8,78 @@ class Shot extends AbstractShot {
     this.ownerId = ownerId;
   }
 
-  insert() {
-    let value = JSON.stringify(this.asJson());
-    return db.insert(
-      `INSERT INTO data (id, deviceid, value)
-       VALUES ($1, $2, $3)`,
-      [this.id, this.ownerId, value]
+  convertAnyDataUrls(client, json, possibleClipsToInsert) {
+    return Promise.all(
+      possibleClipsToInsert.map((clipId) => {
+        let clip = this.getClip(clipId);
+        let imageId = `${this.id}/${clipId}`;
+        let data;
+        try {
+          data = clip.imageBinary();
+        } catch (e) {
+          if (e.message !== "Bad clip URL") {
+            throw e;
+          }
+          // It's already in the db, and the clip has an http url
+          return true;
```

Comment:
This is way over-catching.  We should have a method like `clip.isHttpUrl()` or else test it directly here.  Also we need to account for text clips.